test/position-managers/FeeCollection.t.sol

```diff
@@ -0,0 +1,229 @@
+        vm.startPrank(bob);
+        delta = collect(tokenIdBob, ZERO_BYTES);
+        vm.stopPrank();
+        uint256 balance0BobAfter = currency0.balanceOf(bob);
+        uint256 balance1BobAfter = currency1.balanceOf(bob);
+
+        assertEq(balance0BobBefore, balance0BobAfter);
+        assertEq(uint256(uint128(delta.amount1())), balance1BobAfter - balance1BobBefore);
+        assertTrue(delta.amount1() != 0);
+
+        // position manager holds no fees now
+        assertApproxEqAbs(manager.balanceOf(address(lpm), currency0.toId()), 0, 1 wei);
+        assertApproxEqAbs(manager.balanceOf(address(lpm), currency1.toId()), 0, 1 wei);
+    }
+
+    function test_collect_donate() public {}
+    function test_collect_donate_sameRange() public {}
+
+    /// @dev Alice and Bob create liquidity on the same range, and decrease their liquidity
+    // Even though their positions are the same range, they are unique positions in pool manager.
+    function test_decreaseLiquidity_sameRange_exact() public {
+        // alice and bob create liquidity on the same range [-120, 120]
+        LiquidityRange memory range = LiquidityRange({poolKey: key, tickLower: -120, tickUpper: 120});
+
+        // alice provisions 3x the amount of liquidity as bob
+        uint256 liquidityAlice = 3000e18;
+        uint256 liquidityBob = 1000e18;
+
+        vm.prank(alice);
+        BalanceDelta lpDeltaAlice = mint(range, liquidityAlice, alice, ZERO_BYTES);
+        uint256 tokenIdAlice = lpm.nextTokenId() - 1;
+
+        uint256 aliceBalance0Before = IERC20(Currency.unwrap(currency0)).balanceOf(address(alice));
+        uint256 aliceBalance1Before = IERC20(Currency.unwrap(currency1)).balanceOf(address(alice));
+
+        vm.prank(bob);
+        BalanceDelta lpDeltaBob = mint(range, liquidityBob, bob, ZERO_BYTES);
+        uint256 tokenIdBob = lpm.nextTokenId() - 1;
+
+        uint256 bobBalance0Before = IERC20(Currency.unwrap(currency0)).balanceOf(address(bob));
+        uint256 bobBalance1Before = IERC20(Currency.unwrap(currency1)).balanceOf(address(bob));
+
+        // swap to create fees
+        uint256 swapAmount = 0.001e18;
+        swap(key, true, -int256(swapAmount), ZERO_BYTES); // zeroForOne is true, so zero is the input
+        swap(key, false, -int256(swapAmount), ZERO_BYTES); // move the price back, // zeroForOne is false, so one is the input
+
+        // alice decreases liquidity
+        vm.startPrank(alice);
+        lpm.approve(address(this), tokenIdAlice);
+        decreaseLiquidity(tokenIdAlice, liquidityAlice, ZERO_BYTES);
+        vm.stopPrank();
+
+        uint256 tolerance = 0.000000001 ether;
+
+        // alice has accrued her principle liquidity + any fees in token0
+        assertApproxEqAbs(
+            IERC20(Currency.unwrap(currency0)).balanceOf(address(alice)) - aliceBalance0Before,
+            uint256(int256(-lpDeltaAlice.amount0())) + swapAmount.mulWadDown(FEE_WAD) * 3 / 4,
```

Comment:
why `* 3 / 4`?

Reply:
alice owned 75% of the liquidity so should be entitled to 75% of the fee revenue. updated to make it more clear